File: apps/api/app/schemas/topic.py
from pydantic import BaseModel, ConfigDict


class TopicRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    slug: str
    name: str
    description: str | None
    order_index: int
    parent_topic_id: int | None


class TopicWithSubtopicsRead(TopicRead):
    subtopics: list[TopicRead]

File: apps/web/lib/types/topic.ts
export type TopicRead = {
  id: number;
  slug: string;
  name: string;
  description: string | null;
  order_index: number;
  parent_topic_id: number | null;
};

export type TopicWithSubtopics = TopicRead & {
  subtopics: TopicRead[];
};
